examples/nix-shell/shell.ncl

```diff
@@ -0,0 +1,20 @@
+{
+  inputs = {
+    hello = { input = "nixpkgs" }
+  }
+
+  , output = fun { pkgs, nix, .. } =>
+    { # (optional) name of the shell
+       name = "hello"
+
+       , packages = [
+         pkgs.hello
+       ]
+
+       , shellHook = nix.lib.nix_string_hack [
+          "echo 'Development shell'",
+          pkgs.hello,
+          "/bin/hello"
+       ]
```

Comment:
Actually it can't work with cowsay. `lib.nix` used the declared inputs to pass to Nickel whatever is needed, because JSON is used as an exchange format, everything that is serialized must be forced. So we have to know what to export to JSON in advance. I guess this makes nix_string_hack totally useless in our model?

Reply:
Then I don't understand how it's different from `"hello"` or useful. If I put it into the inputs, can't I just use a native Nickel string:

```
shellHook = m%"
  %{pkgs.cowsay}/bin/cowsay 'Welcome to this nickel-powered shell'
 "%m
```

?

Reply:
Technically I think nickel wouldn’t evaluate that `pkgs.cowsay` _probably_ can’t be directly interpolated to a string (but that’s a detail and we could change that).

But the more fundamental reason preventing this is that having a variable in `inputs` doesn’t mean that it will be an input of every derivation in the file − and it’s actually not added as input unless you explicitely specify it.

As a (really contrieved example), https://github.com/thufschmitt/nickel-nix/tree/compound-shell has an intermediate derivation, and the final shell depends on `perl`, but not the intermediate derivation